### ISSUE TITLE
Clarify that @trpc/server is needed as a dependency in the client

### ISSUE THREAD

**Issue by davazp** [trpc/trpc#1461]
This is not really a bug, but I thought it would help others if this is made a bit more clear. 🙂 

We have a monorepo, in which I installed `@trpc/client` and `@trpc/server` in different packages.

However, I did not add `@trpc/server` to the client.

Because of how the packages are (not) hoisted, this resulted in the client not being able to infer the types correctly and using `any` as output without errors.

The reason is that `@trpc/client` needs the server package to be available. But Typescript did not complain, due to the skipLibCheck option, I guess.

## Ways to improve this

- This should have resulted in an error ideally.

- At least, the instructions on the website could instruct to install `@trpc/server` as a devDependency in the client.

- Another idea would be to have `@trpc/core`, which is maybe more intuitive than having `@trpc/server` as a dependency in the client?

**Comment by davazp** [comment]
It probably did, but it is a large project full of many other peerDependency warnings so I missed it. 😨

**Comment by KATT** [comment]
Feel free to do a tweak anywhere in the docs where you find it suitable - there are edit buttons that take your straight to the markdown editor on GitHub. :)

**Comment by davazp** [comment]
I am not sure if it is worthy complicating the documentation after all for this.

- It is not an issue with `npm`, as it auto-install peer-dependencies.
- yarn/pnpm will warn about them

What if instead the internal packages did not peer-depend on each other but just regularly depend?

As each package depend on the exact same version of the others, this should work, I think?

Otherwise, feel free to close it :-)

**Comment by KATT** [comment]
There's risk of multiple installs with that solution unfortunately, but it's worth exploring further, for sure. Could do it differently for next major if there is a good solution.

**Comment by davazp** [comment]
> There's risk of multiple installs with that solution unfortunately

True, if both `@trpc/client` and `@trpc/server` are installed on the same package, as in the examples, it is true that then we rely on the package manager to hoist them / deduplicate.

Although, that already happens in my setup unfortunately, as the client needs `@trpc/server`,  I have two copies of it. Depending on the package manager, that might or might not be hoisted.

In my case, it is not, so I do have two copies of `@trpc/server`. It seems fine because it's used for the types and they are version.
